node/runtime/src/lib.rs

```diff
@@ -121,24 +123,39 @@ impl balances::Trait for Runtime {
 	type TransferPayment = ();
 }
 
-impl consensus::Trait for Runtime {
-	type Log = Log;
-	type SessionKey = AuthorityId;
-
-	// The Aura module handles offline-reports internally
-	// rather than using an explicit report system.
-	type InherentOfflineReport = ();
-}
-
 impl timestamp::Trait for Runtime {
 	type Moment = u64;
 	type OnTimestampSet = Aura;
 }
 
+parameter_types! {
+	pub const Period: BlockNumber = 10 * MINUTES;
+	pub const Offset: BlockNumber = 0;
+}
+
+type SessionHandlers = (Grandpa,);
+#[cfg(feature = "std")]
+use serde::{Serialize, Deserialize};
+impl_opaque_keys! {
+	pub struct SessionKeys(AuthorityId,);
+}
+
+// NOTE: `SessionHandler` and `SessionKeys` are co-dependent: One key will be used for each handler.
+// The number and order of items in `SessionHandler` *MUST* be the same number and order of keys in
+// `SessionKeys`.
```

Comment:
basically, you'll end up with default keys rather than the actual keys. a macro would be easiest to do. it may be possible to do something cleverer with traits to avoid exposing a macro to the user, but i haven't though about it too deeply.